hydra_agent/agent.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from hydra_agent.redis_proxy import RedisProxy
+from hydra_agent.graphutils_operations import GraphOperations
+from hydra_agent.hydra_graph import InitialGraph
+from hydra_python_core import doc_maker
+from requests import Session
+
+logger = logging.getLogger(__file__)
+
+
+class Agent(Session):
+    def __init__(self, entrypoint_url):
+        self.entrypoint_url = entrypoint_url.strip().rstrip('/')
+        self.redis_proxy = RedisProxy()
+        self.redis_connection = self.redis_proxy.get_connection()
+        self.graph_operations = GraphOperations(entrypoint_url,
+                                                self.redis_proxy)
+        super().__init__()
+        jsonld_api_doc = super().get(self.entrypoint_url + '/vocab').json()
+        self.api_doc = doc_maker.create_doc(jsonld_api_doc)
+        self.initialize_graph()
+
+    def initialize_graph(self):
+        self.graph = InitialGraph()
+        self.redis_connection.delete("apigraph")
+        self.graph.main(self.entrypoint_url, self.api_doc, True)
+        self.redis_connection.sadd("fs:url", self.entrypoint_url)
+
+    def get(self, url):
+        response = self.graph_operations.get_resource(url)
+        if response is not None:
+            return response
+
+        response = super().get(url)
+
+        if response.status_code == 200:
+            self.graph_operations.get_processing(url, response.json())
+
+        return response.json()
+
+    def put(self, url, new_object):
```

Comment:
The agent is missing docstrings and I'll complement some signature types also

Reply:
yes thanks. after that I am ok to merge